ray_data_eval/libsolver/src/solver/environment.rs

```diff
@@ -270,11 +303,29 @@ impl Executor {
             if task.remaining_ticks > 0 {
                 self.timeline.push_str(&task.spec.id);
                 self.timeline.push_str("  ");
+
+                if task.remaining_ticks == task.spec.duration as i32 {
+                    let dur_time = (task.spec.duration as u64) * 500_000;
+                    let event = TraceEvent {
```

Comment:
Make a constructor function for TraceEvent (implement `TraceEvent::new()`) so here we can just do `self.timeline_json.push(TraceEvent::new(...))`